scipy/linalg/_basic.py

```diff
@@ -855,6 +883,10 @@ def solve_circulant(c, b, singular='raise', tol=None,
     if nc != nb:
         raise ValueError(f'Shapes of c {c.shape} and b {b.shape} are incompatible')
 
+    # accommodate empty arrays
+    if b.size == 0:
+        return np.empty_like(b)
```

Comment:
minor issue: this misses the dtype determination. I commented on the corresponding test case with more detail.